nuxeo-platform-video-core/src/main/java/org/nuxeo/ecm/platform/video/tools/VideoConcat.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * (C) Copyright 2016 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Ricardo Dias
+ */
+package org.nuxeo.ecm.platform.video.tools;
+
+import org.apache.commons.io.FilenameUtils;
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.api.blobholder.BlobHolder;
+import org.nuxeo.runtime.api.Framework;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The {@link VideoTool} that joins two or more video blobs.
+ *
+ * @since 8.4
+ */
+public class VideoConcat extends VideoTool {
+
+    public final static String NAME = "concatTool";
+
+    protected final static String VIDEOS_FILE_PATH_PARAM = "listFilePath";
+
+    protected final static String VIDEO_CONCAT_COMMANDLINE = "videoConcat";
+
+    public VideoConcat() {
+        super(VideoConcat.NAME, VIDEO_CONCAT_COMMANDLINE);
+    }
+
+    @Override
+    public Map<String, String> setupParameters(BlobHolder blobHolder, Map<String, Object> parameters) {
+        List<Blob> videos = blobHolder.getBlobs();
+        List<File> sourceFiles = new ArrayList<>();
+        File tempFile;
+        try {
+            String list = "";
+            for (Blob b : videos) {
+                File blobFile = b.getFile();
+                sourceFiles.add(blobFile);
+                list += "file '" + blobFile.getAbsolutePath() + "'\n";
+            }
+            tempFile = Framework.createTempFile("NxVTcv-", ".txt");
+            Framework.trackFile(tempFile, this);
+            Files.write(tempFile.toPath(), list.getBytes());
```

Comment:
Never use `getBytes` without a charset. Presumably UTF-8 here? But it has to be explicit.